### PR TITLE
Issue #343: Add grouped products support

### LINKED ISSUE #343: Support grouped products export

### PR DESCRIPTION
**Pull request by tim-bezhashvyly**
Closes #343.

Maybe there's more than this. Maybe I just overlooked something.

### CODE REVIEW

File: magento-plugin/app/code/community/LizardsAndPumpkins/MagentoConnector/Model/Resource/Catalog/Product/Collection.php [hunk 1]
```diff
@@ -329,25 +329,32 @@ public function loadAssociatedSimpleProductData()
...
+        $select->joinLeft(
+            ['super_link' => $coreResource->getTableName('catalog/product_super_link')],
+            "e.entity_id = super_link.product_id AND super_link.parent_id IN ({$connection->quote($this->productIds)})",
+            []
+        );
+        $select->joinLeft(
+            ['link' => $coreResource->getTableName('catalog/product_link')],
+            "e.entity_id = link.linked_product_id AND link.product_id IN ({$connection->quote($this->productIds)})",
+            []
+        );
```
Comment (Vinai) [inline magento-plugin/app/code/community/LizardsAndPumpkins/MagentoConnector/Model/Resource/Catalog/Product/Collection.php]: I think you only need to join a single table, the `catalog_product_relation` table.
IIRC the `catalog_product_super_link` was there first, when there only where configurable products, and later when other product types where added the former table was added as the "right" one.
The table `catalog_product_relation` was kept for backward compatibility.

The `catalog_product_link` table contains even more information, not only parent/child, but upsells and crossells and related products. I don't think we need that.

I think the right thing to do, if we want to support the composite product types besides configurables, would be to join in only the `catalog_product_relation` table to select the associated simple products.

**Comment by Vinai** [comment]
I assumed you visually confirmed this change doesn't break configurable products in the export?

**Comment by tim-bezhashvyly** [comment]
I did but please don't merge it until the tax class issue is resolved.

**Comment by Vinai** [comment]
I'll hold off with the merge until the tax class issues are resolved as requested.

**Comment by tim-bezhashvyly** [comment]
Can be merged now.

**Pull request merged**